From: Egbert Eich <eich@suse.com>
Date: Tue Nov 27 14:44:14 2018 +0100
Subject: Perl: Escape left curly brace properly in regexps for perl 5.26
Patch-mainline: Upstream
Git-repo: https://github.com/warewulf/warewulf3
Git-commit: 25f1016479392975403d0de207c3fe1fc48f5bcd
References: 

Perl version 5.26 requires left curly brace to be escaped.

Not escaping them will give an error like:
"Unescaped left brace in regex is illegal here in regex; marked by <--
HERE in m/\%{ <-- HERE NETWORK}/"

See also:
https://metacpan.org/pod/release/RJBS/perl-5.22.0/pod/perldelta.pod#A-literal-%22%7B%22-should-now-be-escaped-in-a-pattern

Signed-off-by: Egbert Eich <eich@suse.com>
Signed-off-by: Egbert Eich <eich@suse.de>
---
 provision/lib/Warewulf/Provision/Dhcp/Isc.pm | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)
diff --git a/provision/lib/Warewulf/Provision/Dhcp/Isc.pm b/provision/lib/Warewulf/Provision/Dhcp/Isc.pm
index 4b6de56..43487d5 100644
--- a/provision/lib/Warewulf/Provision/Dhcp/Isc.pm
+++ b/provision/lib/Warewulf/Provision/Dhcp/Isc.pm
@@ -201,9 +201,9 @@ persist()
         return(1);
     }
 
-    $config_template =~ s/\%{IPADDR}/$ipaddr/g;
-    $config_template =~ s/\%{NETWORK}/$network/g;
-    $config_template =~ s/\%{NETMASK}/$netmask/g;
+    $config_template =~ s/\%\{IPADDR}/$ipaddr/g;
+    $config_template =~ s/\%\{NETWORK}/$network/g;
+    $config_template =~ s/\%\{NETMASK}/$netmask/g;
 
     &dprint("Creating DHCPD configuration file header\n");
     $dhcpd_contents .= "# DHCPD Configuration written by Warewulf. Do not edit this file, rather\n";
